index.js

```diff
@@ -5,8 +5,31 @@
  * MIT Licensed
  */
 var promiseExists = typeof Promise === 'function';
-var globalObject = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : self; // eslint-disable-line
-var isDom = 'location' in globalObject && 'document' in globalObject;
+
+/* eslint-disable */
+// See http://stackoverflow.com/a/6930376
+var globalObject;
+try {
+  globalObject = Function('return this')() || (42, eval)('this');
+} catch (e) {
+  globalObject = window;
+}
+/* eslint-enable */
+
+/*
+ * All of these attributes must be available on the global object for the current environment
+ * to be considered a DOM environment (browser)
+ */
+var domIndicatorAttributes = [
+  'location',
+  'document',
+  'navigator',
+  'HTMLElement',
+];
+var isDom = domIndicatorAttributes.every(function inGlobal(attr) {
+  return attr in globalObject;
+});
```

Comment:
Not massively keen on doing these kind of iterations upfront. Would rather see the 4 `attr in globalObject` lines hardcoded than using an array.